src/main.c

```diff
@@ -2408,15 +2409,85 @@ void handle_mouse_input() {
     }
 }
 
-void handle_movement(double dt) {
+/// [issue](https://github.com/Team-10-But-Better/Craft/issues/9)
+/// This function returns the movement speed based on whether the character should teleport.
+///
+///\param teleport : whether the character should teleport
+///\param teleportHeldLastFrame : whether the teleport button was previously held
+///\param groundedMovementSpeed : default movement speed
+
+int getGroundedMovementSpeed(bool teleport, bool *teleportHeldLastFrame, int groundedMovementSpeed)
+{
+    if (teleport)
+    {
+        if (*teleportHeldLastFrame == false){
+            *teleportHeldLastFrame = true;
+            return 600;
+        }
+        *teleportHeldLastFrame = true;
+    }
+    else
+    {
+        *teleportHeldLastFrame = false;
+    }
+
+    return groundedMovementSpeed;
+}
+
+/// [issue](https://github.com/Team-10-But-Better/Craft/issues/69)
+/// This function returns what height the character should be at (depending on if they are crouching)
+/// and changes the pointer that holds whether the character was crouching last frame
+///
+///\param crouch : whether the character should crouch
+///\param crouchingHeldLastFrame : whether the crouch button was held last frame
+
+int getCrouchingHeight(bool crouch, bool *crouchingHeldLastFrame)
+{
+    if(crouch){
+        *crouchingHeldLastFrame = true;
+    }
+    else{
+        *crouchingHeldLastFrame = false;
+    }
+
+    return crouch ? 1 : 2;
+}
+
+/// [issue](https://github.com/Team-10-But-Better/Craft/issues/69)
+/// This function returns whether the character should be crouching
+///
+///\param crouch : whether the character should crouch
+bool isCrouching(bool crouch){
+    return crouch;
+}
+
+void handle_movement(double dt, bool *teleportHeldLastFrame, bool *crouchHeldLastFrame)
```

Comment:
The best way to handle these parameters (teleportHeldLastFrame, crouchHeldLastFrame) would be to put them into a type defined struct that holds all data about buttons held in the last frame. This would reduce the number of parameters in the function and allow for future conditions checking for buttons held in the last frame to be checked easily.